yjit.c

```diff
@@ -43,6 +43,8 @@ enum rstring_offsets {
     RUBY_OFFSET_RSTRING_EMBED_LEN = offsetof(struct RString, as.embed.len),
 };
 
+bool rb_yjit_should_run_wb;
+
```

Comment:
Hmm, why a global variable thooo?

Reply:
I'm not thrilled about the global either, but I'll try to give context.  I had a conversation with @peterzhu2118 and @eightbitraptor yesterday, and we want to be able to replace Ruby's GC with other implementations (say mmtk for example).  As such, I don't want to expose the layout of `rb_objspace` to YJIT.  @XrXr suggested that I write a function that returns the address of the bit where the GC tracks incremental marking, but that seems kind of fraught when it comes to using a 3rd party GC (could they even provide that?).  A callback system where the GC indicates "from now, you must always fire the WB" seemed more reasonable.

Since I didn't want YJIT to dig in to the GC, I wasn't sure where to store this info so I just made a global.  If you know a better place to store the info, I'd be happy to put it there!

Reply:
I understand. There's maybe a question as to whether we should store that variable on the Rust side and expose a Rust method instead @XrXr, but that is a minor point.

Reply:
On second thought, I do see that you read the value in the generated code, so maybe it's easier to do that with a C variable. Would still appreciate input from Alan, but your solution makes sense to me.